config/env.go

```diff
@@ -0,0 +1,35 @@
+package config
+
+// TODO: everything in this file should be either removed or unexported
+import (
+	"os"
+	"strconv"
+	"strings"
+)
+
+// ConfigKeyToEnv gets the env variable name from a given config key
+func ConfigKeyToEnv(s string) string {
+	if upperCaseMatch.MatchString(s) {
+		return s
+	}
```

Comment:
[optional] Some stuff doesn't seem to be covered as reported by codecov here. Just pointing it out because you did a great job with the tests and these seem fairly easy to add.

Reply:
I intentionally left `GetXXX` package-level functions uncovered, since they simply delegate the call to the singleton instance and the intention is to remove them at a later stage. Will try to cover others